libyul/CompilabilityChecker.cpp

```diff
@@ -59,21 +59,13 @@ map<YulString, int> CompilabilityChecker::run(
 			builtinContext,
 			_optimizeStackAllocation
 		);
-		try
-		{
-			transform(*_object.code);
-		}
-		catch (StackTooDeepError const&)
-		{
-			yulAssert(!transform.stackErrors().empty(), "Got stack too deep exception that was not stored.");
-		}
+		transform(*_object.code);
 
-		std::map<YulString, int> functions;
 		for (StackTooDeepError const& error: transform.stackErrors())
-			functions[error.functionName] = max(error.depth, functions[error.functionName]);
-
-		return functions;
+		{
+			unreachableVariables[error.functionName].emplace(error.variable);
+			auto& deficit = stackDeficit[error.functionName];
```

Comment:
```suggestion
			int& deficit = stackDeficit[error.functionName];
```

Reply:
I already have an "auto-removal" commit, but apparently I didn't catch enough of them :-).